File: rentio-client/src/types/review.ts
export type ReviewStatus = "PENDING" | "APPROVED" | "REJECTED"

export interface Review {
  id: number
  userId: number
  userName: string
  flatId: number
  flatAddress?: string // for MyReviews + admin display
  title: string
  content: string
  rating: number
  reviewDate: string
  status: ReviewStatus
}

export interface CreateReviewRequest {
  title: string
  content: string
  rating: number
}

File: rentrix-server/src/main/java/com/rentrix/rentrixserver/entity/Review.java
package com.rentrix.rentrixserver.entity;

import com.rentrix.rentrixserver.entity.constants.ReviewStatus;
import jakarta.persistence.*;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;

import java.time.LocalDate;

@Data
@Entity
@Table(name = "reviews")
public class Review {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "user_id", nullable = false)
	private User user;
	
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "flat_id", nullable = false)
	private Flat flat;
	
	@NotBlank
	private String title;
	
	@NotBlank
	@Lob
	private String content;
	
	@Min(1)
	@Max(10)
	private Integer rating;
	
	private LocalDate reviewDate;
	
	@Enumerated(EnumType.STRING)
	private ReviewStatus status = ReviewStatus.PENDING;
	
}